metadata-ingestion/src/datahub/ingestion/source/identity/azure_ad.py

```diff
@@ -419,18 +419,10 @@ def ingest_ad_users(
         for user_count, datahub_corp_user_snapshot in enumerate(
             datahub_corp_user_snapshots
         ):
-            # Add GroupMembership if applicable
-            if (
-                datahub_corp_user_snapshot.urn
-                in datahub_corp_user_urn_to_group_membership.keys()
-            ):
-                datahub_group_membership = (
-                    datahub_corp_user_urn_to_group_membership.get(
-                        datahub_corp_user_snapshot.urn
-                    )
-                )
-                assert datahub_group_membership
-                datahub_corp_user_snapshot.aspects.append(datahub_group_membership)
+            datahub_corp_user_urn: str = datahub_corp_user_snapshot.urn
+            datahub_group_membership: GroupMembershipClass = datahub_corp_user_urn_to_group_membership[datahub_corp_user_urn]
+            assert datahub_group_membership is not None
```

Comment:
I think this assert is not required

also the "Add GroupMembership" comment was still useful

Reply:
Can do!